eng/common/scripts/Cadl-Project-Generate.ps1

```diff
@@ -0,0 +1,67 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory
+)
+
+$ErrorActionPreference = "Stop"
+. $PSScriptRoot/Helpers/PSModule-Helpers.ps1
+. $PSScriptRoot/common.ps1
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+
+function NpmInstallForProject([string]$workingDirectory) {
+    Push-Location $workingDirectory
+    try {
+        $currentDur = Resolve-Path "."
+        Write-Host "Generating from $currentDur"
+        if (Test-Path "package.json") {
+            Remove-Item -Path "package.json" -Force
+        }
+        if (Test-Path ".npmrc") {
+            Remove-Item -Path ".npmrc" -Force
+        }
+        $replacementPackageJson = "$PSScriptRoot/../../emitter-package.json"
+        Write-Host("Copying package.json from $replacementPackageJson")
+        Copy-Item -Path $replacementPackageJson -Destination "package.json" -Force
+        npm install
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+$emitterName = &$GetEmitterNameFn
+$cadlConfigurationFile = Resolve-Path "$ProjectDirectory/cadl-location.yaml"
+
+Write-Host "Reading configuration from $cadlConfigurationFile"
+$configuration = Get-Content -Path $cadlConfigurationFile -Raw | ConvertFrom-Yaml
+
+$specSubDirectory = $configuration["directory"]
+$innerFolder = Split-Path $specSubDirectory -Leaf
+
+$tempFolder = "$ProjectDirectory/TempCadlFiles"
```

Comment:
If there isn't a good reason to have the Temp folder in the same directory it would be nice come up with some other convention, maybe even the global temp folder. 

If we keep it under the repo we will need to get it added to .gitignore in the other repos as well.

Reply:
It is added to .gitignore and the advantage here is if you want to do some local changes / what if having it right next to your project is convenient